src/seedu/addressbook/AddressBook.java

```diff
@@ -257,17 +250,13 @@ private static void echoUserCommand(String userCommand) {
      * @param args full program arguments passed to application main method
      */
     private static void processProgramArgs(String[] args) {
-        if (args.length >= 2) {
-            showToUser(MESSAGE_INVALID_PROGRAM_ARGS);
-            exitProgram();
-        }
-
         if (args.length == 1) {
             setupGivenFileForStorage(args[0]);
-        }
-
-        if(args.length == 0) {
+        } else if(args.length == 0) {
             setupDefaultFileForStorage();
+        } else {
+            showToUser(MESSAGE_INVALID_PROGRAM_ARGS);
+            exitProgram();
```

Comment:
Do watch out for using else case to represent the "rest" of the case. You may end up allowing cases where you don't want to. "Else If" would be more explicit in setting boundaries.